gems/aws-sigv4/spec/aws-sigv4/signer_spec.rb

```diff
@@ -769,18 +736,38 @@ def initialize(credentials)
 
         context 'expires in' do
           it 'defaults to 15 minutes' do
-            presigned_url = subject.presign_url(request: request)
+            presigned_url = subject.presign_url(
+              request: request, credentials: credentials
+            )
             expect(presigned_url.url.to_s).to include('X-Amz-Expires=900')
           end
 
           it 'allows for expires in override' do
-            presigned_url = subject.presign_url(request: request, expires_in: 1)
+            presigned_url = subject.presign_url(
+              request: request, credentials: credentials, expires_in: 1
+            )
             expect(presigned_url.url.to_s).to include('X-Amz-Expires=1')
           end
 
           it 'raises when expires in is not an integer' do
-            expect { subject.presign_url(request: request, expires_in: 'foo') }
-              .to raise_error(ArgumentError, /number of seconds/)
+            expect do
+              subject.presign_url(
+                request: request, credentials: credentials, expires_in: 'foo'
+              )
+            end.to raise_error(ArgumentError, /number of seconds/)
+          end
+
+          context 'credentials expiration' do
```

Comment:
Ditto about "test duplication" pattern in this spec. Let's ensure expiration works for requests too (if applicable)

Reply:
I dont think its applicable - this is only used for setting the TTL of a presigned url.